### PR TITLE
@types/ws: upgrade to the latest and move to devDependencies

### PR DESCRIPTION
**Pull request by arniu**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,9 +37,10 @@
     "tsd": "^0.11.0"
   },
   "dependencies": {
+    "@types/ws": "^7.2.2",
```
Comment (mcollina) [inline package.json]: I would highly prefer to avoid adding this external dependency for JS devs

Comment (arniu) [reply package.json]: Me too. But the type from `ws` will be `any` for TS devs.

Comment (mcollina) [reply package.json]: They can always install that package if they need it.

File: index.d.ts [hunk 2]
```diff
@@ -82,8 +82,8 @@ declare namespace websocketPlugin {
   }
 
   export interface PluginOptions {
-    handle: (connection: SocketStream) => void;
-    options: WebSocket.ServerOptions;
+    handle?: (connection: SocketStream) => void;
```
Comment (Ethan-Arrowood) [inline index.d.ts]: Why were these changed to optional? Sorry I'm not that familiar with the repo.

Comment (arniu) [reply index.d.ts]: They are optional. If not changed, type checking will report errors.

Comment (Ethan-Arrowood) [reply index.d.ts]: Okay! Then LGTM

**Comment by Eomm** [comment]
@Ethan-Arrowood is semver patch this change?
So I may release it accordingly 👍

**Comment by Ethan-Arrowood** [comment]
Yes this shouldn't break anything

**Comment by mcollina** [comment]
This was done.
